### PR TITLE
Feature/stp 834 acceptance tests

### PR DESCRIPTION
**Pull request by TheSpartan1980**
# Description

Created additional acceptance tests to cover the most commonly used components and actions carried out by users in user research in effort to increase the value and quality of the tests that we currently have. 

## Type of change

- [x] New acceptance tests

# How Has This Been Tested?

- All tests were created using the red-green-refactor cycle
- Tests were run in parallel against the latest version of master in Chrome and headless mode
- Tests were run in a single instance against the latest version of master in Chrome and headless mode

# Checklist:

- [x] My changes do not introduce any new linting errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] I have rebased onto master and there are no code conflicts

### CODE REVIEW

File: smoke-tests/features/step-definitions/linkingPages.steps.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+When("I link ths page to link from the {string}", (linkedPage) => {
+  this.newPageName = "Second page";
+  AddPageSection.linkFrom(linkedPage);
+  AddPageSection.pageTitle.setValue(this.newPageName);
+  AddPageSection.saveBtn.click();
+});
+
+Then("my page is created with a link to the page", () => {
+  this.pageName = this.newPageName.toLowerCase().replace(" ", "-");
```
Comment (jenbutongit) [inline smoke-tests/features/step-definitions/linkingPages.steps.js:22]: This is how we are sanitising for URLs, it's in designer/client/helpers. feel free to lift. If you're comfortable with refactoring it into model/helpers, you can do so or you can do the ol copy + paste and put a `//TODO:` 

```
export function toUrl(title) {
  return `/${(title?.replace(/[^a-zA-Z0-9- ]/g, "") ?? "")
    .trim()
    .replace(/ +/g, "-")
    .toLowerCase()}`;
}
```

File: smoke-tests/features/step-definitions/steps.js [hunk 2]
```diff
@@ -1,63 +1,88 @@
...
-      expect(FormDesignerPage[pageComponent](this.pageName)).toHaveText(
-        "dd/mm/yyyy"
+  switch (componentName) {
+    case "Paragraph":
+      AddComponentPage.paragraphSetText(
+        "You need the vehicle’s number plate (registration number)." +
+          "\n" +
+          "You can see the results as soon as the MOT centre has recorded the test result." +
+          "\n" +
+          "You’ll need the 11-digit number from the vehicle’s log book (V5C) to see the test location."
```
Comment (jenbutongit) [inline smoke-tests/features/step-definitions/steps.js:29]: you can do this with js template literals, which supports multi line so you don't have to do the "+ \n" thing
```
`You need the vehicle’s number plate (registration number). \n
    You can see the results as soon as the MOT centre has recorded the test result. \n
    You’ll need the 11-digit number from the vehicle’s log book (V5C) to see the test location.`
```

File: smoke-tests/features/step-definitions/steps.js [hunk 3]
```diff
@@ -1,63 +1,88 @@
...
-        .to.be.true;
-      break;
   }
 });
 
+Then(
+  "the {string} control is displayed in the {string}",
+  (componentName, pageName) => {
+    const pageComponent = toCamelCase(componentName);
+    console.log("PAGE COMPONENT IS: " + pageComponent);
```
Comment (jenbutongit) [inline smoke-tests/features/step-definitions/steps.js:45]: did you mean to commit console log?

File: smoke-tests/wdio.CI.conf.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+const { config } = require("./wdio.conf.js");
+
+(config.baseUrl = "http://localhost:3000"), // TODO:- Change to the url in the CI environment
```
Comment (jenbutongit) [inline smoke-tests/wdio.CI.conf.js:3]: remind me to help you set this up

**Review by jenbutongit — CHANGES_REQUESTED** [review]

File: smoke-tests/features/step-definitions/steps.js (hunk 3 above)
Comment (TheSpartan1980) [reply smoke-tests/features/step-definitions/steps.js:45]: Removed

File: smoke-tests/features/step-definitions/steps.js (hunk 2 above)
Comment (TheSpartan1980) [reply smoke-tests/features/step-definitions/steps.js:29]: Amended to incorporate template literals.

File: smoke-tests/features/step-definitions/linkingPages.steps.js (hunk 1 above)
Comment (TheSpartan1980) [reply smoke-tests/features/step-definitions/linkingPages.steps.js:22]: These lines of code are used to locate the page by CSS locator as the ID's in the HTML are displayed as follows: id="/first-page"

File: smoke-tests/wdio.CI.conf.js (hunk 4 above)
Comment (TheSpartan1980) [reply smoke-tests/wdio.CI.conf.js:3]: Will do I'll ensure a ticket is created 👍

**Review by jenbutongit — APPROVED** [review]

**Pull request merged**